File: WebAPI/Models/AvailabilityRequestModelCreate.cs
using System.Collections.Generic;

namespace WebAPI.Models
{
    public class AvailabilityRequestModelCreate
    {
        public string Address { get; set; }
        public string Conditions { get; set; }
        public int RequestStateId { get; set; }
        public int UserId { get; set; }
        public List<AvailableTimeModelForCreateRequest> AvailableTimeModelForCreateRequests { get; set; }
    }
}

File: WebAPI/ClientApp/src/app/models/AvailabilityRequestModelCreate.ts
import { AvailableTimeModelForCreateRequest } from "./AvailableTimeModelForCreateRequest";

export class AvailabilityRequestModelCreate {
    address: string;
    conditions: string;
    requestStateId: number;
    userId: number;
    availableTimeModelForCreateRequests: AvailableTimeModelForCreateRequest[];

    public constructor(_address: string, _conditions: string, _requestStateId: number, _userId: number, _availableTimeModelForCreateRequests: AvailableTimeModelForCreateRequest[]) {
        this.address = _address;
        this.conditions = _conditions;
        this.requestStateId = _requestStateId;
        this.userId = _userId;
        this.availableTimeModelForCreateRequests = _availableTimeModelForCreateRequests;
    }
}
